File: BaseConverter/BaseConverter.cpp
#include <iostream>
#include <cmath>
#include <string>
using namespace std;

string convert(string, int, int);
int toDecimal(string, int);
string toBase(int, int);
char maxCheck(int);

int main(){

    cout << "Enter the number you're converting: ";
    string inpNum;
    cin >> inpNum;
    int base = 0;
    do{
        cout << "What base is this number in?\nEnter a number (2-35) here: ";
        cin >> base;
    }while(base < 2 || base > 35);
    int newBase = 0;
    do{
    cout << "What base is this number being converted to?\nEnter a number here: ";
    cin >> newBase;
    }while(base < 2 || base > 35);
    cout << convert(inpNum, base, newBase) << endl;
    return 0;
}

string convert(string num, int oldbase, int newbase){
    int deci = toDecimal(num, oldbase);
    return toBase(deci, newbase);
}
int toDecimal(string num, int oldbase){
    if(oldbase == 10)
        return stoi(num);
    int deci = 0;
    for(int x = num.size()-1; x >= 0; x--){
        int power = num.size()-1-x;
        deci+=((num[x]-'0')*pow(oldbase, power));
    }
    return deci;
}
string toBase(int deci, int newbase){
    if(newbase == 10)
        return to_string(deci);
    string toRet = "";
    int div = deci, mod = 0;
    while(div != 0){
        mod = div%newbase;
        div = div/newbase;
        toRet=maxCheck(mod)+toRet;
    }
    return toRet;
}
char maxCheck(int num){
    if(num > 9)
	return('7'+num);
    return '0'+num;
}

File: BaseConverter/BaseConverter.py
def convert(num, oldbase, newbase):
    deci = toDecimal(num, oldbase)
    return toBase(deci, newbase)

def toDecimal(num, oldbase):
    if oldbase == 10:
        return int(num)
    deci = 0
    for x in range(len(num)-1, 0):
        power = len(num)-1-x
        deci+=((num[x]-ord('0'))**(oldbase, power))
    return deci

def toBase(deci, newbase):
    if newbase == 10:
        return deci
    toRet = ""
    div = deci
    while div != 0:
        mod = div%newbase
        div = int(div/newbase)
        toRet=maxCheck(mod)+toRet
    return toRet

def maxCheck(num):
    if num > 9:
	    return chr(ord('7')+num)
    return chr(ord('0')+num)

inpNum = input("Enter the number you're converting: ")
base = 0
while base < 2 or base > 35:
    base = int(input("What base is this number in?\nEnter a number (2-35) here: "))
newBase = 0
while newBase < 2 or newBase > 35:
    newBase = int(input("What base is this number being converted to?\nEnter a number here: "))

print(convert(inpNum, base, newBase))